SimDataFormats/GeneratorProducts/src/LHERunInfoProduct.cc

```diff
@@ -206,104 +206,102 @@ bool HeaderLess::operator() (const LHERunInfoProduct::Header &a,
 }
 
 bool LHERunInfoProduct::isTagComparedInMerge(const std::string& tag) {
-        return !(tag == "" || tag.find("Alpgen") == 0 || tag == "MGGridCard" || tag == "MGGenerationInfo" || tag=="MGRunCard" || tag == "mgruncard" || tag=="MadSpin" || tag=="madspin");
+        return !(tag == "" || tag.find("Alpgen") == 0 || tag == "MGGridCard" || tag=="MGRunCard" || tag == "mgruncard" || tag=="MadSpin" || tag=="madspin");
 }
 
 bool LHERunInfoProduct::mergeProduct(const LHERunInfoProduct &other)
 {
-	if (heprup_.IDBMUP != other.heprup_.IDBMUP ||
+
+  if (heprup_.IDBMUP != other.heprup_.IDBMUP ||
 	    heprup_.EBMUP != other.heprup_.EBMUP ||
 	    heprup_.PDFGUP != other.heprup_.PDFGUP ||
 	    heprup_.PDFSUP != other.heprup_.PDFSUP ||
 	    heprup_.IDWTUP != other.heprup_.IDWTUP) {
+        
 	  return false;	
 	}
 
-	bool compatibleHeaders = headers_ == other.headers_;
+	bool compatibleHeaders = (headers_ == other.headers_);
 
-	// try to merge different, but compatible headers
+	// try to merge not equal but compatible headers (i.e. different iseed)
 	while(!compatibleHeaders) {
-		// okay, something is different.
+		// okay, something is not the same.
 		// Let's try to merge, but don't duplicate identical headers
 		// and test the rest against a whitelist
 
 		std::set<Header, HeaderLess> headers;
 		std::copy(headers_begin(), headers_end(),
 		          std::inserter(headers, headers.begin()));
 
+    // make a list of headers contained in the second file
+    std::vector<std::vector<std::string> > runcard_v2;
+    std::vector<std::string> runcard_v2_header;
+    for(std::vector<LHERunInfoProduct::Header>::const_iterator
+      header2 = headers_.begin();
+      header2 != headers_.end(); ++header2) {
+        // fill a vector with the relevant header tags that can be not equal but sill compatible
+        if(header2->tag()=="MGRunCard" || header2->tag() == "mgruncard" || header2->tag() == ""){
+          runcard_v2.push_back(header2->lines());
+          runcard_v2_header.push_back(header2->tag());
+        }
+    }
+    
+    // loop over the headers of the original file
 		bool failed = false;
 		for(std::vector<LHERunInfoProduct::Header>::const_iterator
 					header = other.headers_begin();
 		    header != other.headers_end(); ++header) {
+          
 			if (headers.count(*header)) {
 				continue;
 			}
-
+            
+      if(header->tag()=="MGRunCard" || header->tag() == "mgruncard" || header->tag() == ""){
+        bool header_compatible = false;
+        for (unsigned int iter_runcard = 0; iter_runcard < runcard_v2.size(); iter_runcard++){
+          // this is relevant for MG5_aMG@NLO in LO mode
+          std::vector<std::string> runcard_v1 = header->lines();
+          runcard_v1.erase( std::remove_if( runcard_v1.begin(), runcard_v1.end(), [](const std::string& x) { return x.find("iseed") != std::string::npos; } ), runcard_v1.end() );        
```

Comment:
StackOverflow suggests this list of `find`s could be replaced with one `find_if`:
http://stackoverflow.com/questions/19952155/pass-multiple-strings-to-the-stringfind-function/19952425#19952425

The `if` statements with multiple string equality comparisons could be replaced similarly.